tuplex/python/src/PythonContext.cc

```diff
@@ -211,9 +211,22 @@ namespace tuplex {
                 // first get how many bytes are required
                 size_t requiredBytes = baseRequiredBytes;
                 if(varLenField) {
-                    for(int j = 0; j < numTupleElements; ++j)
-                        if(typeStr[j] == 's')
-                            requiredBytes += PyUnicode_GET_SIZE(PyTuple_GetItem(obj, j)) + 1; // +1 for '\0'
+                    bool nonConforming = false;
+                    for(int j = 0; j < numTupleElements; ++j) {
+                        if (typeStr[j] == 's') {
+                            auto tupleItem = PyTuple_GetItem(obj, j);
+                            if (PyUnicode_Check(tupleItem)) {
+                                requiredBytes += PyUnicode_GET_SIZE(PyTuple_GetItem(obj, j)) + 1; // +1 for '\0'
```

Comment:
don't call PyTuple_GetItem here again, use the one from before and adjust refcount to be correct. https://docs.python.org/3.9/c-api/refcounting.html (check doc for PyTuple as well, maybe toy example to get correct refcount here :) )